llvm/lib/Target/AMDGPU/SIRegisterInfo.td

```diff
@@ -1108,54 +1108,64 @@ class RegOrImmOperand <RegisterClass RegClass, string OperandTypeName>
     let ParserMatchClass = RegImmMatcher<!subst("_Deferred", "", NAME)>;
 }
 
+// Should be in sync with the OperandSemantics defined in SIDefines.h
+def OperandSemantics {
+  int INT = 0;
+  int FP16 = 1;
+  int BF16 = 2;
+  int FP32 = 3;
+  int FP64 = 4;
```

Comment:
I believe for now it only matters for 16-bit literal but we can keep them just in case.